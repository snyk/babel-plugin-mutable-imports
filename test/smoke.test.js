import { transform } from '@babel/core';

function runPlugin(code) {
  return transform(code, {
    babelrc: false,
    filename: 'test.js',
    plugins: [__dirname + '/../lib/plugin.js'],
  });
}

describe('smoke testing', () => {
  it('transforms just one line', () => {
    // language=JavaScript
    const code =
      'Object.defineProperty(exports, "foo",' +
      '{ enumerable: true,' +
      ' get: function () { return b_1.foo; } });';

    expect(runPlugin(code).code).toMatchSnapshot();
  });
});
