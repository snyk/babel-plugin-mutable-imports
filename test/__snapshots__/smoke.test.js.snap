// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke testing transforms just one line 1`] = `undefined`;
